platforms/access-bridge/src/services/stripe-service.ts

```diff
@@ -0,0 +1,60 @@
+import Stripe from 'stripe';
+
+import { STRIPE_SECRET } from '../app-config.js';
+
+export type StripeProduct = Stripe.Product & {
+  prices: Stripe.Price[];
+};
+
+/**
+ * Service class responsible for interacting with the Stripe API to fetch products.
+ */
+export class StripeService {
+  private stripe: Stripe;
+
+  constructor() {
+    this.stripe = new Stripe(STRIPE_SECRET, {
+      apiVersion: '2024-06-20',
+    });
+  }
+
+  /**
+   * Retrieves Stripe products with prices and filters them based on the provided productIds.
+   * @param productIds The array of product IDs to filter the products.
+   * @returns A Promise resolving to an array of filtered ProductsWithMetadata objects.
+   */
+  async getProductsWithPrices(productIds: string[]): Promise<StripeProduct[]> {
+    if (!productIds.length) {
+      return [];
+    }
+
+    // Fetch all products (pagination might be necessary for large datasets)
+    const allProducts = await this.stripe.products.list();
+    const filteredProducts = allProducts.data.filter((product) => productIds.includes(product.id));
+
+    if (filteredProducts.length === 0) {
+      return [];
+    }
+
+    // Retrieve prices for each filtered product
+    const productsWithPrices = await Promise.all(
+      filteredProducts.map(async (product) => {
+        try {
+          const prices = await this.stripe.prices.list({ product: product.id });
+          return {
+            ...product,
+            prices: prices.data,
+          };
+        } catch (priceError) {
+          console.error(`Failed to fetch prices for product ${product.id}:`, priceError);
+          return {
+            ...product,
+            prices: [], // Return an empty array if price retrieval fails
```

Comment:
What's the idea behind this error path? Does it result in us showing viewers a plan that they can't buy? Maybe it's better to just let this fail the entire flow and we can show a message to viewers that there was a problem fetching plans with a refresh button.

Reply:
I've updated to not return the product if the price fails. So in the end if the products array is empty, the FE will handle it.